File: engine/engine-model/src/main/scala/tools/packet/ChangeChannel.scala
package tools.packet

import java.net.InetAddress

import net.opcodes.SendOpcode

class ChangeChannel(private var _inetAddress: InetAddress, private var _port: Int) extends PacketInput {
  def inetAddress: InetAddress = _inetAddress

  def port: Int = _port

  override def opcode(): SendOpcode = SendOpcode.CHANGE_CHANNEL
}

File: engine/engine-model/src/main/java/tools/packet/ChangeChannel.java
package tools.packet;

import java.net.InetAddress;

import net.opcodes.SendOpcode;

public record ChangeChannel(InetAddress inetAddress, Integer port) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CHANGE_CHANNEL;
   }
}